src/lib.rs

```diff
@@ -166,9 +169,13 @@ pub trait DataNftMint:
 
         let treasury_address = self.treasury_address().get();
 
+        let bond_amount = self.get_bond_amount_for_lock_period(lock_period_sec);
+
         if price >= BigUint::zero() {
```

Comment:
What's the point of this if? Can't you just do:

require!(
                payment.amount >= &price + &bond_amount,
                ERR_NOT_ENOUGH_FUNDS
            );

Reply:
If payment is 0, it shouldn't send funds to treasury address, if the check is not here will be somewhere else in the code ..

Reply:
Alright. Makes sense. Then I guess the check for equality can be gotten out of the if.

Also, as a cool thing (not mandatory to do) you can user direct_non_zero to catch the case in which the transfer is zero